File: checkarray_sorted.py
def check(self, nums: list[int]) -> bool:
    if(all(nums[i] <= nums[i+1] for i in range(len(nums) - 1))):
        return True
    rp=0
    for i in range(len(nums)-1):
        if(nums[i]>nums[i+1]):
            rp=i+1
    x=(len(nums))-rp
    b=[0]*(len(nums))
    for i in range(len(nums)):
        b[(i+x)%(len(nums))]=nums[i]
    if(all(b[i] <= b[i+1] for i in range(len(b) - 1))):
        return True
    else:
        return False

File: check_array_sorted.java
public class check_array_sorted {
    public static boolean check(int[] nums) {
            int n=nums.length;
            int dp=0;
            for(int i=0;i<n;i++){
                if(nums[i]>nums[(i+1)%n]){
                    dp+=1;
                }
                if(dp>1){
                    return false;
                }
            }
            return true;
        }
        public static void main(String[]args){
            //example, main class
            int [] nums={3,4,5,1,2};
            check(nums);
    }
}
